main/src/main/java/org/mobilitydata/gtfsvalidator/validator/TripTravelSpeedValidator.java

```diff
@@ -0,0 +1,131 @@
+ *
+ * <p>Time complexity: <i>O(n * p)</i>, where <i>n</i> is the number of records in
+ * <i>stop_times.txt</i> and <i>p</i> is the number of points in a trip shape.
+ *
+ * <p>Generated notice: {@link FastTravelBetweenStopsNotice}.
+ */
+@GtfsValidator
+public class TripTravelSpeedValidator extends FileValidator {
+  private static final double METER_PER_SECOND_TO_KMH_CONVERSION_FACTOR = 3.6D;
+  @Inject GtfsTripTableContainer tripTable;
+  @Inject GtfsStopTimeTableContainer stopTimeTable;
+  @Inject GtfsStopTableContainer stopTable;
+
+  @Override
+  public void validate(NoticeContainer noticeContainer) {
+    for (Entry<String, List<GtfsStopTime>> entry :
+        Multimaps.asMap(stopTimeTable.byTripIdMap()).entrySet()) {
+      List<FastTravelBetweenStopsNotice> noticesForTrip = checkSpeedAlongTrip(entry);
+      for (FastTravelBetweenStopsNotice notice : noticesForTrip) {
+        noticeContainer.addValidationNotice(notice);
+      }
+    }
+  }
+
+  /**
+   * Calculates the travel speed between {@code GtfsStopTime}. Generates a {@code
+   * FastTravelBetweenStopsNotice} if the travel speed between {@code GtfsStopTimes} is greater than
+   * 150 km/h.
+   *
+   * @param tripStopTimes the list of {@code GtfsStopTimes} sorted by stop_times.stop_sequence for a
+   *     {@code GtfsTrip}
+   * @return the list of {@code FastTravelBetweenStopsNotice} generated when checking speed travel
+   *     along the given trip
+   */
+  private List<FastTravelBetweenStopsNotice> checkSpeedAlongTrip(
+      Entry<String, List<GtfsStopTime>> tripStopTimes) {
+    var previousStopsData =
+        new Object() {
+          final List<Integer> accumulatedStopSequence = new ArrayList<>();
+          GtfsTime departureTime = null;
+          double latitude;
+          double longitude;
+          // used to accumulate distance between stops with same arrival and departure
+          // times
+          int accumulatedDistanceMeter = 0;
+        };
+    List<FastTravelBetweenStopsNotice> notices = new ArrayList<>();
+    for (GtfsStopTime stopTime : tripStopTimes.getValue()) { // prepare data for current iteration
+      GtfsStop currentStop = stopTable.byStopId(stopTime.stopId());
+      double currentStopLat = currentStop.stopLat();
+      double currentStopLon = currentStop.stopLon();
+      GtfsTime currentArrivalTime = stopTime.arrivalTime();
+      double distanceFromPreviousStopMeter =
+          GeospatialUtil.distanceInMeterBetween(
+              previousStopsData.latitude,
+              previousStopsData.longitude,
+              currentStopLat,
+              currentStopLon);
+      boolean sameArrivalAndDeparture = false;
+      if (previousStopsData.departureTime != null && currentArrivalTime != null) {
+        sameArrivalAndDeparture = currentArrivalTime.equals(previousStopsData.departureTime);
+        if (!sameArrivalAndDeparture) {
+          int durationSecond =
+              currentArrivalTime.getSecondsSinceMidnight()
```

Comment:
If there is a `StopTimeWithArrivalBeforePreviousDepartureTimeNotice` for this trip this `TripTravelSpeedValidator` will produce some strange results. You may want to detect this here and bail on this trip if so.

I'm also curious as to what the real-world performance benefits would be if we consolidated this and some of the other rules like `StopTimeWithArrivalBeforePreviousDepartureTimeNotice` that loop through all of stop_times.txt, as we're currently doing this loop multiple times for each `*Validator` class, and stop_times.txt can be very large. Looks like `StopTimeWithDepartureBeforeArrivalTimeNotice` and `StopTimeWithArrivalBeforePreviousDepartureTimeNotice` are already consolidated within `StopTimeArrivalAndDepartureTimeValidator`.